cupy/cuda/cupy_thrust.cu

```diff
@@ -43,6 +44,102 @@ public:
 };
 
 
+/* ------------------------------------ Minimum boilerplate to support complex numbers ------------------------------------ */
+// We need a specialized operator< here in order to match the NumPy behavior:
+// "The sort order for complex numbers is lexicographic. If both the real and imaginary parts are non-nan then the order is
+// determined by the real parts except when they are equal, in which case the order is determined by the imaginary parts.
+//
+// In numpy versions >= 1.4.0 nan values are sorted to the end. The extended sort order is:
+//     Real: [R, nan]
+//     Complex: [R + Rj, R + nanj, nan + Rj, nan + nanj]
+// where R is a non-nan real value. Complex values with the same nan placements are sorted according to the non-nan part if
+// it exists. Non-nan values are sorted as before."
+// Ref: https://docs.scipy.org/doc/numpy/reference/generated/numpy.sort.html
+
+__host__ __device__ inline bool isnan(const cuComplex& z) {
+    return isnan(z.x) || isnan(z.y);
+}
+
+__host__ __device__ inline bool isnan(const cuDoubleComplex& z) {
+    return isnan(z.x) || isnan(z.y);
+}
```

Comment:
The current definition in cupy of `isnan` for complex numbers does not work?

Reply:
As noted above, I couldn't figure out how to use CuPy's complex headers, so no.